src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -106,7 +115,7 @@ private static float parseTime(Element testCase) {
         return 0.0f;
     }
 
-    CaseResult(SuiteResult parent, Element testCase, String testClassName, boolean keepLongStdio) {
+    CaseResult(SuiteResult parent, Element testCase, String testClassName, boolean keepLongStdio, @CheckForNull String archiveId) {
```

Comment:
there is no need for this ctor signature change as the property is available in the parent.  (but see my above general comment)